### PR TITLE
LPS-156125,LPS-156123,LPS-156122

### PR DESCRIPTION
**Pull request by rodrigopaulino**
Related ticket:
https://issues.liferay.com/browse/LPS-156125

### CODE REVIEW

**Comment by rodrigopaulino** [comment]
@carolmariaabb this PR contains commits from LPS-156123 and LPS-156122 also. They were already reviewed by you and their related pull request already was forwarded to Brian's repo. Therefore, before forwarding this, wait to see if the other PR was merged without changes.
Thx

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by rodrigopaulino** [comment]
ci:test:relevant

File: modules/apps/object/object-service/service.xml [hunk 1]
```diff
@@ -446,6 +446,7 @@
 		<column name="name" type="String" />
 		<column name="reverse" type="boolean" />
 		<column name="type" type="String" />
+		<column name="parameterObjectFieldId" type="long" />
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/service.xml:449]: Shouldn't the `parameterObjectFieldId` be right before the `reverse` column?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v3_15_0/ObjectRelationshipUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+/**
+ * @author Rodrigo Paulino
+ */
+public class ObjectRelationshipUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (!hasColumn("ObjectRelationship", "parameterObjectFieldId")) {
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v3_15_0/ObjectRelationshipUpgradeProcess.java:26]: We don't need to use the `hasColumn` anymore. The `alterTableAddColumn` method is now doing this check (see [LPS-147889](https://issues.liferay.com/browse/LPS-147889))

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 3]
```diff
@@ -72,17 +76,29 @@
...
 	public ObjectRelationship addObjectRelationship(
 			long userId, long objectDefinitionId1, long objectDefinitionId2,
 			String deletionType, Map<Locale, String> labelMap, String name,
 			String type)
 		throws PortalException {
 
+		return addObjectRelationship(
+			userId, objectDefinitionId1, objectDefinitionId2, deletionType,
+			labelMap, name, type, null);
+	}
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:90]: Do we really need to keep this method?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 4]
```diff
@@ -308,13 +324,23 @@ public List<ObjectRelationship> getObjectRelationships(
...
-	@Indexable(type = IndexableType.REINDEX)
 	@Override
 	public ObjectRelationship updateObjectRelationship(
 			long objectRelationshipId, String deletionType,
 			Map<Locale, String> labelMap)
 		throws PortalException {
 
+		return updateObjectRelationship(
+			objectRelationshipId, deletionType, labelMap, null);
+	}
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:336]: Do we really need to keep this method?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 5]
```diff
@@ -561,6 +607,43 @@ private void _validate(
...
+		}
+
+		SystemObjectDefinitionMetadata systemObjectDefinitionMetadata =
+			_systemObjectDefinitionMetadataTracker.
+				getSystemObjectDefinitionMetadata(objectDefinition.getName());
+
+		if (systemObjectDefinitionMetadata != null) {
+			restContextPath =
+				systemObjectDefinitionMetadata.getRESTContextPath();
+		}
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:640]: Can we add this code inside an `else` statement?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 6]
```diff
@@ -72,17 +76,29 @@
...
+			userId, objectDefinitionId1, objectDefinitionId2, deletionType,
+			labelMap, name, type, null);
+	}
+
+	@Indexable(type = IndexableType.REINDEX)
+	@Override
+	public ObjectRelationship addObjectRelationship(
+			long userId, long objectDefinitionId1, long objectDefinitionId2,
+			String deletionType, Map<Locale, String> labelMap, String name,
+			String type, Long parameterObjectFieldId)
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:97]: I'm still confused about the `parameterObjectFieldId` coming after the `type`.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 7]
```diff
@@ -423,6 +463,11 @@ private ObjectRelationship _addObjectRelationship(
 		objectRelationship.setReverse(reverse);
 		objectRelationship.setType(type);
 
+		if (parameterObjectFieldId != null) {
+			objectRelationship.setParameterObjectFieldId(
+				parameterObjectFieldId);
+		}
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:470]: same as above

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 8]
```diff
@@ -478,7 +524,7 @@ else if (Objects.equals(
 
 	private void _validate(
 			long objectDefinitionId1, long objectDefinitionId2, String name,
-			String type)
+			String type, Long parameterObjectFieldId)
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:528]: same as above

File: modules/apps/object/object-service/service.xml (hunk 1 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-service/service.xml:449]: As discussed offline, I'll follow previous precedent.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v3_15_0/ObjectRelationshipUpgradeProcess.java (hunk 2 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v3_15_0/ObjectRelationshipUpgradeProcess.java:26]: Nice. Didn't know that! Thanks, Carol!

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java (hunk 4 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:336]: As discussed offline, I'll follow previous precedent.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java (hunk 6 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:97]: As discussed offline, I'll follow previous precedent.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java (hunk 7 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:470]: As discussed offline, I'll follow previous precedent.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java (hunk 8 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:528]: As discussed offline, I'll follow previous precedent.

**Comment by rodrigopaulino** [comment]
ci:close

**Pull request closed without merging**